docs/advanced_examples/KNearestNeighbors.ipynb

```diff
@@ -0,0 +1,431 @@
+    "\n",
+    "# Fit both the Concrete ML and its equivalent float estimators on clear data\n",
+    "concrete_knn, sklearn_model = concrete_knn.fit_benchmark(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Compile the model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Saved in 47\n",
+      "Compilation time: 2.56 seconds\n"
+     ]
+    }
+   ],
+   "source": [
+    "# The compilation step aims to:\n",
+    "# - convert the quantized model to its FHE equivalent\n",
+    "# - create an executable operation graph\n",
+    "# - check the operation graph's compatibility with FHE\n",
+    "# - compute the maximum bit-width needed for model execution\n",
+    "# - determine cryptographic parameters necessary for generating secret keys and evaluation keys\n",
+    "time_begin = time.time()\n",
+    "circuit = concrete_knn.compile(X)\n",
+    "print(f\"Compilation time: {time.time() - time_begin:.2f} seconds\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Generating a key for an 8-bit circuit\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f\"Generating a key for an {circuit.graph.maximum_integer_bit_width()}-bit circuit\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Key generation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Key generation time: 829.81 seconds\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Note that this step may be time-consuming for circuits exceeding 8-bits\n",
+    "time_begin = time.time()\n",
+    "circuit.client.keygen(force=False)\n",
+    "print(f\"Key generation time: {time.time() - time_begin:.2f} seconds\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Inference with Concrete ML:\n",
+    "\n",
+    "a. __clear__: inference on unencrypted quantized data, without any FHE execution \n",
+    "\n",
+    "b. __Execution in FHE__: inference on encrypted data, using actual FHE execution\n",
+    "\n",
+    "c. __Simulation__: inference on unencrypted quantized data, without secure FHE execution, while simulating the p_error failure probability. For further information, please consult: [TODO]()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# scikit-learn inference\n",
+    "predict_sklearn = sklearn_model.predict(X_test)\n",
+    "score_sklearn = (predict_sklearn == y_test).mean()"
```

Comment:
also maybe it's better to use sklearn's accuracy score for computing all scores, looks cleaner imo

Reply:
good idea